pkg/csi/service/wcpguest/controller.go

```diff
@@ -288,25 +305,44 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("ControllerPublishVolume: called with args %+v", *req)
+	// Check whether the request is for a block or file volume
+	isFileVolumeRequest := common.IsFileVolumeRequest(ctx, []*csi.VolumeCapability{req.GetVolumeCapability()})
+
 	err := validateGuestClusterControllerPublishVolumeRequest(ctx, req)
 	if err != nil {
 		msg := fmt.Sprintf("Validation for PublishVolume Request: %+v has failed. Error: %v", *req, err)
 		log.Error(msg)
 		return nil, status.Errorf(codes.Internal, msg)
 	}
 
+	// File volumes support
+	if isFileVolumeRequest {
+		// Check the feature state for file volume support
+		if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FileVolume) {
+			// Feature is disabled on the cluster
+			return nil, status.Error(codes.InvalidArgument, "File volume not supported.")
+		}
+		return controllerPublishForFileVolume(ctx, req, c)
+	}
+	// Block volumes support
+	return controllerPublishForBlockVolume(ctx, req, c)
+}
+
+// controllerPublishForBlockVolume is a helper mthod for handling ControllerPublishVolume request for Block volumes
+func controllerPublishForBlockVolume(ctx context.Context, req *csi.ControllerPublishVolumeRequest, c *controller) (
```

Comment:
This could be a method under `controller` struct.

Reply:
Keeping the methods as util/helper methods within the `wcpguest` package for now